Content.Server/_DV/Abilities/Kitsune/KitsuneSystem.cs

```diff
@@ -45,21 +56,52 @@ private void OnPolymorphed(Entity<KitsuneComponent> ent, ref PolymorphedEvent ar
         {
             if (!TryComp<FoxfireComponent>(fireUid, out var foxfire))
                 continue;
-            foxfire.Kitsune = args.NewEntity;
+            foxfire.Kitsune = newEntity;
             Dirty(fireUid, foxfire);
         }
 
-        //Transfer Accesses
-        var accessItems = _reader.FindPotentialAccessItems(ent);
-        var accesses = _reader.FindAccessTags(ent, accessItems);
-        EnsureComp<AccessComponent>(args.NewEntity);
-        _access.TrySetTags(args.NewEntity, accesses);
+        // Code after this point will not run when reverting to human form.
+        if(HasComp<KitsuneFoxComponent>(oldEntity))
+            return;
+
+        // Transfer Accesses
+        var accessItems = _reader.FindPotentialAccessItems(oldEntity);
+        var accesses = _reader.FindAccessTags(oldEntity, accessItems);
+        EnsureComp<AccessComponent>(newEntity);
+        _access.TrySetTags(newEntity, accesses);
 
-        //Transfer factions
-        if (TryComp<NpcFactionMemberComponent>(ent, out var factions))
+        // Transfer factions
+        if (TryComp<NpcFactionMemberComponent>(oldEntity, out var factions))
         {
-            EnsureComp<NpcFactionMemberComponent>(args.NewEntity);
-            _faction.AddFactions(args.NewEntity, factions.Factions);
+            EnsureComp<NpcFactionMemberComponent>(newEntity);
+            _faction.AddFactions(newEntity, factions.Factions);
+        }
+
+        // Transfer equipped item speed modifiers
+        if (TryComp<MovementSpeedModifierComponent>(newEntity, out var movementComp))
```

Comment:
Is there a reason to do TryComp here instead of Ensure comp? Sounds reasonable to me to just force slowdown, regardless of what you turn into, no?